src/lib.rs

```diff
@@ -323,87 +363,102 @@ impl AsyncHttpRangeReader {
             .parse()
             .map_err(|_err| AsyncHttpRangeReaderError::ContentLengthMissing)?;
 
-        // Allocate a memory map to hold the data
+        let builder = self
+            .url(head_response.url().clone())
+            .content_length(content_length as usize);
+
+        Ok(builder)
+    }
+
+    pub fn from_tail_response(
+        self,
+        tail_response: Response,
+    ) -> Result<Self, AsyncHttpRangeReaderError> {
+        let content_range = ContentRange::parse(
+            tail_response
+                .headers()
+                .get(reqwest::header::CONTENT_RANGE)
+                .ok_or(AsyncHttpRangeReaderError::ContentRangeMissing)?
+                .to_str()
+                .map_err(|_err| AsyncHttpRangeReaderError::ContentRangeMissing)?,
+        );
+        let (start, finish, complete_length) = match content_range {
+            ContentRange::Bytes(ContentRangeBytes {
+                first_byte,
+                last_byte,
+                complete_length,
+            }) => (first_byte, last_byte, complete_length),
+            _ => return Err(AsyncHttpRangeReaderError::HttpRangeRequestUnsupported),
+        };
+
+        let requested_range =
+            SparseRange::from_range(complete_length - (finish - start)..complete_length);
+
+        // Configure the initial state of the streamer.
+        let mut streamer_state = StreamerState::default();
+        streamer_state
+            .requested_ranges
+            .push(complete_length - (finish - start)..complete_length);
+        let builder = self
+            .url(tail_response.url().clone())
+            .initial_tail_response(Some((tail_response, start)))
+            .requested_range(requested_range)
+            .content_length(complete_length as usize)
+            .streamer_state(streamer_state);
+        Ok(builder)
+    }
+
+    pub fn build(self) -> Result<AsyncHttpRangeReader, AsyncHttpRangeReaderBuilderError> {
+        let Some(url) = self.url else {
+            return Err(AsyncHttpRangeReaderBuilderError::MissingUrl);
+        };
+
+        if self.content_length == 0 {
+            return Err(AsyncHttpRangeReaderBuilderError::InvalidContentLength);
+        }
```

Comment:
If I understand it correctly this means you have to call either `.from_head_response` or `.from_tail_response`. If either of those are required lets make those the methods that create the builder. E.g. 

```rust
AsyncHttpRangeReaderBuilder::from_tail_response(url, response)
```

Reply:
Otherwise `AsyncHttpRangeReaderBuilder::new().build()` would result in a runtime error. Would be nicer if a that would be caught at compile time.